end_to_end_test/test/values_serializer_test.dart

```diff
@@ -60,6 +61,33 @@ void main() {
     test('can be deserialized', () {
       expect(serializers.deserialize(serialized), data);
     });
+
+    test('has good error message on deserialize with wrong primitive type', () {
+      final incorrectSerialized = [
+        'CompoundValue',
+        'simpleValue',
+        [
+          'anInt',
+          1,
+          'aString',
+          'two',
+        ],
+        'validatedValue',
+        [
+          'anInt',
+          'foo',
+        ],
+      ];
+      expect(
+          () => serializers.deserialize(incorrectSerialized),
+          throwsA(isErrorContaining(
+              "Deserializing '[CompoundValue, simpleValue, "
```

Comment:
Maybe missing some new lines? Error looks too collapsed to make sense of it IMO.

Reply:
I think it reads okay as a single line. I agree it's not terribly clear mashed into separate lines like this for the test :)

Thanks.